examples/threading/threading.c

```diff
@@ -2,15 +2,51 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <errno.h>
+#include <time.h>
 
 // Optional: use these functions to add debug or error prints to your application
-#define DEBUG_LOG(msg,...)
-//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+//#define DEBUG_LOG(msg,...)
+#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
 #define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
 
 void* threadfunc(void* thread_param)
 {
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
 
+    int ret = usleep(thread_func_args->wait_to_obtain_ms * 1000);
+
+    if (ret == -1) {
+        ERROR_LOG("Error in usleep with code: %d", errno);
+        thread_func_args->thread_complete_success = false;
+        thread_func_args->error_status = 1;
+        return thread_param;
+    } 
+
+    int rc = pthread_mutex_lock(thread_func_args->mutex);
+    if (rc != 0) {
+        ERROR_LOG("Error in mutex lock with code: %d", errno);
+        thread_func_args->thread_complete_success = false;
+        thread_func_args->error_status = 1;
+        return thread_param;
+    }
+    ret = usleep(thread_func_args->wait_to_release_ms * 1000);
+    if (ret == -1) {
+        ERROR_LOG("Error in usleep with code: %d", errno);
+        thread_func_args->thread_complete_success = false;
+        thread_func_args->error_status = 1;
+        return thread_param;
+    } 
+    rc = pthread_mutex_unlock(thread_func_args->mutex);
+    if (rc != 0) {
+        ERROR_LOG("Error in mutex unlock with code: %d", errno);
+        thread_func_args->thread_complete_success = false;
+        thread_func_args->error_status = 1;
+        return thread_param;
+    }
+    
+    if (thread_func_args->error_status == 0)
```

Comment:
@GautamaGandhi does `error_status` need to be a member of the structure?  Or could this just be a local variable?

Reply:
@dwalkes yes, this could be a local variable as well and the same checks can be applied.